internal/compare/compare.go

```diff
@@ -121,22 +123,37 @@ func (p *Pass) service(from, to *compile.ServiceSpec) {
 	if to == nil {
 		// Service was deleted, which is not backwards compatible.
 		p.Report(Diagnostic{
-			File:    filepath.Base(from.File), // toModule could have been deleted.
-			Message: fmt.Sprintf("deleting service %q", from.Name),
+			FilePath: filepath.Base(from.File), // toModule could have been deleted.
+			Message:  fmt.Sprintf("deleting service %q", from.Name),
 		})
 
 		return
 	}
+	file := p.getRelativePath(from.File)
 	for n := range from.Functions {
-		p.function(to.Functions[n], n, filepath.Base(from.File), from.Name)
+		p.function(to.Functions[n], n, file, from.Name)
 	}
 }
 
-func (p *Pass) function(to *compile.FunctionSpec, fn string, file string, service string) {
+// getRelativePath returns a relative path to a file or
+// fallbacks to file name for cases when it was deleted.
+func (p *Pass) getRelativePath(filePath string) string {
+	file, err := filepath.Rel(p.GitDir, filePath)
+	if err != nil {
+		// If a file was deleted, then we will not be able to
+		// find a full path to it.
```

Comment:
did you mean to say we will not be able to find a relative path to it?

Reply:
yup yup. thanks